src/modules/esl-panel-group/core/esl-panel-group.ts

```diff
@@ -238,8 +268,19 @@ export class ESLPanelGroup extends ESLBaseElement {
   protected _onBeforeShow(e: CustomEvent): void {
```

Comment:
I just want to clarify:)
We put an underscore in the method name because it is an event handler? (because as I know protected methods don't need it)

Reply:
Private notation in JS is a very interesting thing... It's not only a visibility modifier, but more often, it is the answer to the question: "Is it safe for me to call or use this from outside? (if I don't well known with the inner implementation)"

Here are 2 cases:
1) We definitely can't say it is something public, and we can allow to use it without context what's inside
2) Another thing is that we try not to conflict with the user in case he is going to extend the class using inheritance. So just because it's a listener, it could often be the case...

And I totally understand your question it's something discussable, for example, we have `onShow`, `onHide` lifecycle methods in ESLTogglable (without private notation), but at the same time, we can't say it's something that should be called from the outside...

Reply:
@ala-n thanks a lot for the detailed answer!

Reply:
Always welcome, @yadamskaya :star: